### DISCUSSION TITLE
ESP8266 + (ens160+aht2x)

### DISCUSSION THREAD

**Discussion by oasi2019** [arendst/Tasmota#20444] [Q&A]
buongiorno malgrado abbia modificato il firmware aggiungendo le righe di programma dovute nel file non mi visualizza i valori dell aht2x cosa posso fare ??

  You should have received a copy of the GNU General Public License
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

#ifndef _USER_CONFIG_OVERRIDE_H_
#define _USER_CONFIG_OVERRIDE_H_

#ifndef USE_AHT2x
#define USE_AHT2x       // [I2cDriver43] Enable AHT20/AM2301B instead of AHT1x humidity and temperature sensor (I2C address 0x38) (+0k8 code)
#endif

#ifndef USE_ENS16x
#define USE_ENS16x         
#endif

/*******************

**Comment by barbudor** [comment]
This is an English only forum
Please edit and translate your question

**Comment by HofeBY** [comment]
@oasi2019

If Console Command "i2cdriver" don´t show driver 43,
so you have to compile your own version of "tasmota.bin" with included driver "AHT20"
Solution for Windows Systems:
Download&install https://code.visualstudio.com/
In VSC add extension SW "PlatformIO IDE"
download&uncompress Binary from https://github.com/arendst/Tasmota
first compile tasmota(ESP8285/ESP8266) and tasmota32(ESP32) will download additional Software & build environment
.
study ...\tasmota\my_user_config.h for possible Drivers & also unwanted Drivers !
.
add wished Driver support in ...\tasmota\user_config_override.h
.
remove unwished Drivers in ...\tasmota\include\tasmota_configurations.h (ESP8285/ESP8266) or tasmota_configurations_ESP32.h (ESP32) mybe Driver by uncomment "#define USE_VEML6070" to "//#define USE_VEML6070"
or disable driver 12 later by command "I2CDriver12 0"

Maybe try out my german compiled Version as an example https://github.com/HofeBY/Tasmota-Firmware

https://github.com/HofeBY/Tasmota-Firmware/blob/main/TasmotaESP32-13.3.0.2-dev-byHofeBY.zip

uncompress and install in right position.

There are also my configuration files for TM13.3.0.2 ESP32 that you can use as an example.
in ...\tasmota\user_config_override.h
feel free to cange it to your needs.
replace "MY_LANGUAGE de_DE" to your natural speach "MY_LANGUAGE it_IT"
also change "LATITUDE" and "LONGITUDE" to your nearby position (for correct sunrise and sunset)
maybe you also want tochange to your own "NTP_ServerX"

dont forget to add your additional drivers nearby AHTx one
#undef USE_ENS16x
#define USE_ENS16x

.
Now build "tasmota" or "tasmota32" depending to your CPU
Install ...\build_output\firmware\tasmota.bin or tasmota32.bin to your device
.
done

don´t forget to write if you get your own functional version

**Reply by barbudor** [reply]
@HofeBY please edit your message to make it readable
